packages/testlab/src/http-server-config.ts

```diff
@@ -38,19 +33,19 @@ export type HostPort = {
  */
 export function givenHttpServerConfig<T extends HttpOptions | HttpsOptions>(
   customConfig?: T,
-): HostPort & T {
+): T {
   const defaults = {
     host: '127.0.0.1',
     port: 0,
     protocol: undefined,
   };
-  const config = Object.assign({}, defaults, customConfig);
+  const config = {...defaults, ...customConfig};
   if (config.host === undefined) config.host = defaults.host;
   if (config.port === undefined) config.port = defaults.port;
   if (isHttpsConfig(config)) {
     setupTlsConfig(config);
   }
-  return config;
+  return config as T;
```

Comment:
At the first sight, your change seems to be towards worse. 
- At the moment, the return type is described as having host, port and any custom config. No explicit cast is needed when returning the merged config.
- With your changes in place, `HostPort` part is gone - are you sure `T` will always include those two fields? Also an explicit cast is needed here.

Let's try to find a cleaner solution that will not require explicit casts.